app/client/keybase/keybase_test.go

```diff
@@ -0,0 +1,467 @@
+	require.Equal(t, kp, KeyPair{})
+}
+
+func TestKeybase_DeleteKeyWrongPassphrase(t *testing.T) {
+	db, err := initDB()
+	defer db.Stop()
+	require.NoError(t, err)
+
+	err = db.ImportFromString(testKey.String(), testPassphrase)
+	require.NoError(t, err)
+
+	_, err = db.GetPrivKey(testKey.Address().String(), testPassphrase)
+	require.NoError(t, err)
+
+	err = db.Delete(testKey.Address().String(), testNewPassphrase)
+	require.ErrorIs(t, err, ErrorWrongPassphrase)
+}
+
+func TestKeybase_SignMessage(t *testing.T) {
+	db, err := initDB()
+	defer db.Stop()
+	require.NoError(t, err)
+
+	pk, err := createTestKeyFromString(testPrivString)
+	require.NoError(t, err)
+
+	err = db.ImportFromString(testPrivString, testPassphrase)
+	require.NoError(t, err)
+
+	privKey, err := db.GetPrivKey(pk.Address().String(), testPassphrase)
+	require.NoError(t, err)
+
+	txBz, err := hex.DecodeString(testTx)
+	require.NoError(t, err)
+
+	signedMsg, err := db.Sign(privKey.Address().String(), testPassphrase, txBz)
+	require.NoError(t, err)
+
+	verified, err := db.Verify(privKey.Address().String(), txBz, signedMsg)
+	require.NoError(t, err)
+	require.Equal(t, verified, true)
+}
+
+func TestKeybase_SignMessageWrongPassphrase(t *testing.T) {
+	db, err := initDB()
+	defer db.Stop()
+	require.NoError(t, err)
+
+	pk, err := createTestKeyFromString(testPrivString)
+	require.NoError(t, err)
+
+	err = db.ImportFromString(testPrivString, testPassphrase)
+	require.NoError(t, err)
+
+	privKey, err := db.GetPrivKey(pk.Address().String(), testPassphrase)
+	require.NoError(t, err)
+
+	txBz, err := hex.DecodeString(testTx)
+	require.NoError(t, err)
+
+	signedMsg, err := db.Sign(privKey.Address().String(), testNewPassphrase, txBz)
+	require.ErrorIs(t, err, ErrorWrongPassphrase)
+	require.Nil(t, signedMsg)
+}
+
+func TestKeybase_ExportString(t *testing.T) {
+	db, err := initDB()
+	defer db.Stop()
+	require.NoError(t, err)
+
+	err = db.ImportFromString(testPrivString, testPassphrase)
+	require.NoError(t, err)
+
+	privStr, err := db.ExportPrivString(testAddr, testPassphrase)
+	require.NoError(t, err)
+	require.Equal(t, privStr, testPrivString)
+}
+
+func TestKeybase_ExportJSON(t *testing.T) {
+	db, err := initDB()
+	defer db.Stop()
+	require.NoError(t, err)
+
+	err = db.ImportFromString(testPrivString, testPassphrase)
+	require.NoError(t, err)
+
+	jsonStr, err := db.ExportPrivJSON(testAddr, testPassphrase)
+	require.NoError(t, err)
+
+	err = db.Delete(testAddr, testPassphrase)
+	require.NoError(t, err)
+
+	err = db.ImportFromJSON(jsonStr, testPassphrase)
+	require.NoError(t, err)
+
+	privKey, err := db.GetPrivKey(testAddr, testPassphrase)
+	require.NoError(t, err)
+	require.Equal(t, privKey.Address().String(), testAddr)
+	require.Equal(t, privKey.String(), testPrivString)
+}
+
+func initDB() (Keybase, error) {
+	db, err := NewKeybaseInMemory("")
+	if err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+func createTestKey() (crypto.PrivateKey, error) {
+	return crypto.GeneratePrivateKey()
```

Comment:
Optional: It's not a must here but look into `keygenerator.GetInstance()`. @deblasis put it together to create test key generation deterministic which can be very important for tests. 

Less so for these, but something to keep in mind.

Reply:
Have changed key generation to use this in the tests